android/java/org/chromium/chrome/browser/crypto_wallet/fragments/dapps/AddSwitchChainNetworkFragment.java

```diff
@@ -0,0 +1,253 @@
+        FragmentNavigationItemAdapter adapter = new FragmentNavigationItemAdapter(this, mTabTitles);
+        viewPager.setAdapter(adapter);
+        new TabLayoutMediator(tabLayout, viewPager,
+                (tab, position) -> tab.setText(mTabTitles.get(position).getTitle()))
+                .attach();
+        Button btnAddSwitchNetwork = view.findViewById(R.id.fragment_add_switch_eth_chain_btn_sign);
+        if (mPanelType == ADD_ETHEREUM_CHAIN) {
+            btnAddSwitchNetwork.setText(R.string.approve);
+            SpannableString addChainHeading = Utils.createSpannableString(
+                    getString(R.string.brave_wallet_allow_add_network_description),
+                    getString(R.string.brave_wallet), new ClickableSpan() {
+                        @Override
+                        public void onClick(@NonNull View widget) {
+                            TabUtils.openUrlInNewTab(false, Utils.BRAVE_SUPPORT_URL);
+                            TabUtils.bringChromeTabbedActivityToTheTop(getActivity());
+                        }
+                    }, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
+
+            TextView tvAddChainTitle = view.findViewById(R.id.fragment_add_switch_chain_tv_title);
+            tvAddChainTitle.setText(R.string.brave_wallet_allow_add_network_heading);
+
+            TextView addChainDesc = view.findViewById(R.id.fragment_add_switch_chain_tv_text);
+            addChainDesc.setMovementMethod(LinkMovementMethod.getInstance());
+            addChainDesc.setText(addChainHeading);
+
+        } else if (mPanelType == SWITCH_ETHEREUM_CHAIN) {
+            btnAddSwitchNetwork.setText(R.string.brave_wallet_allow_change_network_button);
+        }
+        btnAddSwitchNetwork.setOnClickListener(v -> {
+            if (mPanelType == ADD_ETHEREUM_CHAIN) {
+                processAddChainRequest(mNetworkInfo, true);
+                processNextAddNetworkRequest();
+            } else if (mPanelType == SWITCH_ETHEREUM_CHAIN) {
+                processSwitchChainRequest(mSwitchChainRequest, true);
+                processNextSwitchChainRequest();
+            }
+        });
+        Button btnAddSwitchCancel =
+                view.findViewById(R.id.fragment_add_switch_eth_chain_btn_cancel);
+        btnAddSwitchCancel.setOnClickListener(v -> {
+            if (mPanelType == ADD_ETHEREUM_CHAIN) {
+                processAddChainRequest(mNetworkInfo, false);
+                processNextAddNetworkRequest();
+            } else if (mPanelType == SWITCH_ETHEREUM_CHAIN) {
+                processSwitchChainRequest(mSwitchChainRequest, false);
+                processNextSwitchChainRequest();
+            }
+        });
+        return view;
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        // finishNativeInitialization may be called before the view creation
```

Comment:
that's actually a good point, I will revisit what I did for sign message as it fetches only on `finishNativeInitialization`